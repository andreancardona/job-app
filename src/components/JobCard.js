import React from 'react';
import { useHistory } from 'react-router-dom';
import CurrentDate from '../utils/CurrentDate';

const JobCard = (props) => {

  let history = useHistory();
  
  return (
    props.jobs.map((job) => (
      <div className="job-card" key={job.id} tabIndex="0">
        <div className="job-title-location">
          <div className="title" aria-label={"title" + job.title}>{job.title}</div>
          <div className="location" aria-label={"location" + job.location} >{job.location}</div>
        </div>
        <div className="job-post-details">
          <div className="posted" aria-label={"posted" + job.posted} >{job.posted ? job.posted : <CurrentDate />}</div>
          <div className="sponsorship" aria-label={"sponsorship" + job.sponsorship} >{job.sponsorship}</div>
          <div className="status" aria-label={"status" + job.status} >{job.status}</div>
          <button className="edit-button" tabIndex="0" aria-label="edit job" onClick={() => {props.editJob(job) || history.push("/job-form")}} >
            Edit
          </button>
        </div>
      </div>
    ))
  )
}

export default JobCard;