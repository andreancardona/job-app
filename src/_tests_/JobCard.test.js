import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { axe } from 'jest-axe';
import JobCard from '../components/JobCard';

// Initial list of jobs
const jobs = [
  { id: 1, title: 'Product Manager', location: 'Austin, TX', posted: '10/04/2020', sponsorship: 'Free', status: 'Open' },
  { id: 2, title: 'CEO', location: 'Austin, TX', posted: '12/30/2020', sponsorship: 'Sponsored', status: 'Paused'},
  { id: 3, title: 'Software Engineer', location: 'Seattle, WA', posted: '11/18/2020', sponsorship: 'Free', status: 'Closed' },
]

describe('The <JobCard /> component', () => {
  test('JobCard renders with JobCard props', () => {
    const tree = renderer.create(<JobCard jobs={jobs}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });


  test('JobCard should not fail any accessibility tests', async () => {
    const { container } = render(<JobCard jobs={jobs}/>);
    expect(await axe(container)).toHaveNoViolations();
  })
});
