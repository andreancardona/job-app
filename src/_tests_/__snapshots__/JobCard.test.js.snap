// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The <JobCard /> component JobCard renders with JobCard props 1`] = `
Array [
  <div
    className="job-card"
    tabIndex="0"
  >
    <div
      className="job-title-location"
    >
      <div
        aria-label="titleProduct Manager"
        className="title"
      >
        Product Manager
      </div>
      <div
        aria-label="locationAustin, TX"
        className="location"
      >
        Austin, TX
      </div>
    </div>
    <div
      className="job-post-details"
    >
      <div
        aria-label="posted10/04/2020"
        className="posted"
      >
        10/04/2020
      </div>
      <div
        aria-label="sponsorshipFree"
        className="sponsorship"
      >
        Free
      </div>
      <div
        aria-label="statusOpen"
        className="status"
      >
        Open
      </div>
      <button
        aria-label="edit job"
        className="edit-button"
        onClick={[Function]}
        tabIndex="0"
      >
        Edit
      </button>
    </div>
  </div>,
  <div
    className="job-card"
    tabIndex="0"
  >
    <div
      className="job-title-location"
    >
      <div
        aria-label="titleCEO"
        className="title"
      >
        CEO
      </div>
      <div
        aria-label="locationAustin, TX"
        className="location"
      >
        Austin, TX
      </div>
    </div>
    <div
      className="job-post-details"
    >
      <div
        aria-label="posted12/30/2020"
        className="posted"
      >
        12/30/2020
      </div>
      <div
        aria-label="sponsorshipSponsored"
        className="sponsorship"
      >
        Sponsored
      </div>
      <div
        aria-label="statusPaused"
        className="status"
      >
        Paused
      </div>
      <button
        aria-label="edit job"
        className="edit-button"
        onClick={[Function]}
        tabIndex="0"
      >
        Edit
      </button>
    </div>
  </div>,
  <div
    className="job-card"
    tabIndex="0"
  >
    <div
      className="job-title-location"
    >
      <div
        aria-label="titleSoftware Engineer"
        className="title"
      >
        Software Engineer
      </div>
      <div
        aria-label="locationSeattle, WA"
        className="location"
      >
        Seattle, WA
      </div>
    </div>
    <div
      className="job-post-details"
    >
      <div
        aria-label="posted11/18/2020"
        className="posted"
      >
        11/18/2020
      </div>
      <div
        aria-label="sponsorshipFree"
        className="sponsorship"
      >
        Free
      </div>
      <div
        aria-label="statusClosed"
        className="status"
      >
        Closed
      </div>
      <button
        aria-label="edit job"
        className="edit-button"
        onClick={[Function]}
        tabIndex="0"
      >
        Edit
      </button>
    </div>
  </div>,
]
`;
